### PR TITLE
Proper sql logging

### PR DESCRIPTION
**Pull request by BridgeAR**
This has to be merged before #3265 is possible to be merged.

### CODE REVIEW

File: test/integration/promise.test.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ var chai = require('chai')
 
 chai.config.includeStack = true;
 
-describe(Support.getTestDialectTeaser('Promise'), function() {
+describe.skip(Support.getTestDialectTeaser('Promise'), function() {
```
Comment (mickhansen) [inline test/integration/promise.test.js]: skip?

Comment (BridgeAR) [inline test/integration/promise.test.js]: WIP and in my opinion most of the promise tests are not necessary after merging #3265.
I'll also create a PR soon to remove the event-emitter and the query-chainer and deprecate the migrator.

Comment (mickhansen) [inline test/integration/promise.test.js]: Not necessary, but no reason they should break at the moment.

Comment (mickhansen) [inline test/integration/promise.test.js]: query-chainer needs to be deprecated before being removed aswell.

Comment (BridgeAR) [inline test/integration/promise.test.js]: The query-chainer is already deprecated:

  /**
   * The sequelize query chainer allows you to run several queries, either in parallel or serially, and attach a callback that is called when all queries are done.
   *
   * @deprecated The query chainer is deprecated, and due for removal soon. Please use [promises](API-Reference-Promise) instead!
   * @class QueryChainer
   */

Comment (mickhansen) [inline test/integration/promise.test.js]: Does it log a warning when using it? If so we can remove it for 2.1, if not we need to add the warning and then we can remove in 2.2.

**Comment by BridgeAR** [comment]
There are tests and options missing for some functions of the query-interface (like renameColumn).

**Comment by mickhansen** [comment]
LGTM, restarting build. Squash + remove .skip? :)

**Comment by mickhansen** [comment]
Looks good. But build is very broken.

**Comment by BridgeAR** [comment]
This is still WIP, as tehre are quite a few functions in the query-interface where the logging is not implemented yet. I'll introduce options in these functions as discussed in #3265.
The build is broken right now because I couldn't find the missing logging for the select query while using .setInstance. I guess I'll have that fixed tomorrow after refactoring all the other query-interface functions.

**Comment by BridgeAR** [comment]
Seems like the implementation of `instance.removeAssociation` with a belongs-to-many association is broken. The select query is executed twice before the delete statement.

I'll fix that tomorrow.

**Comment by BridgeAR** [comment]
I fixed the instance.removeAssociation firing the select query twice. I hope you do not mind my implementation.

**Comment by janmeier** [comment]
I don't mind the way you implemented it - it actually makes the code a bit cleaner in my mind, or at least more DRY :)

At some point I think `.remove` should just use destroy internally, but that's not the subject of this PR. That would also make it more resistent to concurrency issues, where new join table records are inserted between the select and insert statements

**Comment by BridgeAR** [comment]
Filter is now used. I guess this is ready to merge if it's LGTY.

File: changelog.md [hunk 2]
```diff
@@ -12,6 +17,11 @@
     + inflection@1.6.0
     + lodash@3.5.0
     + validator@3.34
+    + generic-pool@2.2.0
+- [INTERNALS] Updated devDependencies.
```
Comment (mickhansen) [inline changelog.md]: Updating the changelog for a previous version?

File: lib/associations/belongs-to-many.js [hunk 3]
```diff
@@ -415,41 +415,54 @@ module.exports = (function() {
...
+            return newAssociatedObject;
+          });
+        }
+
+        if (options.remove) {
+          oldAssociatedObjects = newAssociatedObjects;
+          newAssociatedObjects = [];
+          oldAssociatedObjects.filter(function(association) {
+            // Determine if this is an association we want to remove
+            return Utils._.find(oldAssociatedObjects, function(oldAssociatedObject) {
```
Comment (mickhansen) [inline lib/associations/belongs-to-many.js]: Won't this always be true? You're matching an array against itself as far as i can tell?

File: lib/query-chainer.js [hunk 4]
```diff
@@ -12,6 +18,8 @@ module.exports = (function() {
   var QueryChainer = function(emitters) {
     var self = this;
 
+    deprecated('The query chainer is deprecated, and due for removal soon. Please use promises (http://sequelize.readthedocs.org/en/latest/api/promise/) instead!');
```
Comment (mickhansen) [inline lib/query-chainer.js]: `due for removal in 2.2`

File: lib/query-interface.js [hunk 5]
```diff
@@ -45,12 +48,13 @@ module.exports = (function() {
...
       raw: true,
-      type: this.sequelize.QueryTypes.SELECT
+      type: this.sequelize.QueryTypes.SELECT,
+      logging: false
     }, options || {});
 
     var showSchemasSql = self.QueryGenerator.showSchemasQuery();
 
-    return this.sequelize.query(showSchemasSql, options).then(function(schemaNames) {
+    return this.sequelize.query(showSchemasSql, undefined, options).then(function(schemaNames) {
```
Comment (mickhansen) [inline lib/query-interface.js]: `options` is fine as the second parameter to `query`, dont need to pass undefined to callee.

File: lib/query-interface.js [hunk 6]
```diff
@@ -245,13 +248,13 @@ module.exports = (function() {
     var skip = options.skip || [];
     return self.showAllTables().then(function(tableNames) {
       if (self.sequelize.options.dialect === 'sqlite') {
-        return self.sequelize.query('PRAGMA foreign_keys;').then(function(result) {
+        return self.sequelize.query('PRAGMA foreign_keys;', undefined, options).then(function(result) {
```
Comment (mickhansen) [inline lib/query-interface.js]: Don't need to add undefined, goes for all the other instances too.

**Comment by mickhansen** [comment]
Looks good overall, just a few comments.

File: changelog.md (hunk 2 above)
Comment (BridgeAR) [inline changelog.md]: Well, these were missing in the changelog. I know it's not nice to change these after releasing but I guess it's always better for people looking things up later on.

File: lib/query-interface.js (hunk 5 above)
Comment (BridgeAR) [inline lib/query-interface.js]: That's just something I personally prefer. Do you want me to change it again?

File: lib/associations/belongs-to-many.js (hunk 3 above)
Comment (BridgeAR) [inline lib/associations/belongs-to-many.js]: Good catch. I did not sleep enough and did not have enough coffee till now^^

Comment (mickhansen) [inline lib/associations/belongs-to-many.js]: Also the filter isn't being assigned to anything :D

File: changelog.md (hunk 2 above)
Comment (mickhansen) [inline changelog.md]: Ah alright :)
devDependencies don't matter much for a changelog, but `generic-pool` is a good add. We'll leave it in.

**Comment by mickhansen** [comment]
Looks good, just need a few squashes and it should be good to go.

**Pull request merged**